client/state/help/selectors.js

```diff
@@ -1,8 +1,16 @@
 /**
  * Internal dependencies
  */
+import { getPrimarySiteId } from 'state/selectors';
+import { getRawSite } from 'state/sites/selectors';
 import createSelector from 'lib/create-selector';
 
 export const getHelpSelectedSiteId = createSelector(
 	state => state.help.selectedSiteId
 );
+
+export const getHelpSelectedSite = ( state ) => {
+	const siteId = getHelpSelectedSiteId( state ) || getPrimarySiteId( state );
```

Comment:
I think you should be using `getSelectedOrPrimarySiteId` here instead of `getPrimarySiteId`. Without the form will always default to the users primary site instead of the site they currently have selected